src/c++/perf_analyzer/data_loader.cc

```diff
@@ -138,12 +138,12 @@ DataLoader::ReadDataFromJSON(
   char readBuffer[65536];
   rapidjson::FileReadStream fs(data_file, readBuffer, sizeof(readBuffer));
 
-  fclose(data_file);
-
   rapidjson::Document d{};
   const unsigned int parseFlags = rapidjson::kParseNanAndInfFlag;
   d.ParseStream<parseFlags>(fs);
 
+  fclose(data_file);
```

Comment:
Can we add a test for this? I wonder why the `L0_perf_analyzer` didn't catch this issue.

Reply:
There are already tests that use input data. The reason a test won't reliably work is because I think sometimes a FILE* can be accessed after a file is closed. A test's failure would be dependent on whether that same file descriptor gets reopened for a different file, which is really hard to control.

Reply:
A lot of the filesystem management for PA needs to be revised to use more modern C++ methods (e.g. std::ifstream, which automatically closes based on scope).